TFT/src/User/API/Printing.c

```diff
@@ -388,14 +388,13 @@ void printAbort(void)
         request_M0();  // M524 is not supportet in reprap firmware
       }
 
-      setDialogText(LABEL_SCREEN_INFO, LABEL_BUSY, LABEL_BACKGROUND, LABEL_BACKGROUND);
-      showDialog(DIALOG_TYPE_INFO, NULL, NULL, NULL);
-
-      do
+      if (infoHost.printing)
```

Comment:
2 Questions about "consistent coding"

1. at least in this file the explicit bool pattern if (boolparam == true) is used. Shouldn't it be same here?
2. at the current state (&hostPrintingConditionCallback == infoHost.printing) should be true always. Wouldn't it be more "future safe" to condtion on if(&hostPrintingConditonCallback == true)?

If I understood the idea of this added if right, it is in order to prevent the "Dialog" calls to happen if loopprocess() will not happen. (compare to old implementations comment on loopProcess in a bottom controled loop instead of a head controled one)
If this old comment is necessary, this implementation will fail on the very day, where &hostPrintingConditionalCallback is false while infoHost.printing is true for whatever reason, as the new implementation of calling loopprocess() is head controled.
At current state this cannot happen, but in future implementations it could happen.

Reply:
1) for boolean types the most used standard style (in the code) is the compact form. Obviuosly the standard form you reported is also correct and it could make sense to uniform it in all the code in the future
2) it's better the curent implementation

that code is correct. It initially check and avoid to display the "busy" message and to iterate on looProcess if the host is not printing anymore. Otherwise it will display the message and wait for the end of the print from the host

Reply:
Thank you for your reply.

about topic 2, that the new implementation is better and correct I saw.

The question I have is, at the current state the values of:
`infoHost.printing `
and
`&hostPrintingConditionCallback `
are identical, because
`bool hostPrintingConditionCallback(void)`
just wraps `infoHost.printing`.

so my question / implied suggestion is only a "refactoring", that could prevent issues in future implementations, if the implementation of hostPrintingConditionCallback(void) changes for whatever reason.

I think this line:
`if (&hostPrintingConditionCallback)`
is better readable and represents the intend better.

--- you may stop reading here, if you dont need closer explaination ---

To explain it:
Let's asume in a future implementation there is a "virtual" condition which allows following state:
`(infoHost.printing == true && &hostPrintingConditionCallback == false) == true`
then it would run into the displaying, but not do the till now exspected "one time looping" as the implementation of
`void loopProcessToCondition(bool (*conditionCallback)(void))`
now is a "head controled" instead of "bottom controled" loop.

The question implied is, if it maybe correct that in the case of above mentioned condition (printer is printing, but the callback functions says it's not needed to wait, that the Dialog is set, but thecall to
`loopProcess()`
does still not happen

Reply:
although it will  not change in the future, your suggestion is correct so I will change the code